### PR TITLE
Enable ieflx_opt=3 for water temperature heat flux from ocean

### PR DESCRIPTION
**Pull request by wlin7**
Internal energy carried by precipitation is accounted for in the ocean,
but not in the atmosphere. This implementation provides the option to
distribute globally uniformly the apparent heat flux from the ocean due
to the disparity in treating precipitation water temperature.

[BFB] for existing F and B configurations, i.e., ieflx=0 or 2.

### CODE REVIEW

**Comment by wlin7** [comment]
WIP: waiting until companion PRs for MPAS-O and cime are merged.

**Review by jonbob — APPROVED** [review]
Looks good from a visual inspection, and a five-year test with these changes looks good

**Comment by oksanaguba** [comment]
@wlin7  --my understanding is that instead of Tsurf or TBOT some other temperature from the ocean will be used in option (3) of influx. What is the difference between Tsurf and that new temperature? thanks!

**Review by golaz — APPROVED** [review]
Based on visual inspection.

**Review by kaizhangpnl — APPROVED** [review]
It seems that the heat flux calculation is done in the ocean model. Everything else looks good to me.

**Comment by wlin7** [comment]
> From @kaizhangpnl : It seems that the heat flux calculation is done in the ocean model. Everything else looks good to me.

Yes, @kaizhangpnl . It is computed in mpas-o and imported to atm via coupler.

> From @oksanaguba : my understanding is that instead of Tsurf or TBOT some other temperature from the ocean will be used in option (3) of influx. What is the difference between Tsurf and that new temperature?

Deferring to @jonbob and @mattdturner . Jon, Mat, can you explain how it is computed? Thanks.

**Comment by jonbob** [comment]
At LANL we are rich in Mat/Matts -- for this one we need @maltrud

**Comment by singhbalwinder** [comment]
Since this PR is approved by all and the changes look good to me too, is it okay if I merge this to next? Do we want to add a test for this new option?

**Comment by wlin7** [comment]
> At LANL we are rich in Mat/Matts -- for this one we need @maltrud

Yes, we always know. Thanks, @jonbob . I didn't realize I pick a Matt when github popped up the choices.

**Comment by whannah1** [comment]
Per our slack conversation, you can remove the changes to the MMF version of camsrfexch.F90 since that duplicate file will be deleted by PR #4145

**Comment by golaz** [comment]
@oksanaguba : replying to your comment above. The temperature is the same (Tsurf = SST over ocean), but the key difference is that we are only doing the calculation for water leaving/entering the ocean. Previously, we were computing it for all the water leaving/entering the atmosphere. As a result, we were overestimating the magnitude of the correction (and possibly compensating for the missing SW term in the ocean).

**Comment by oksanaguba** [comment]
@golaz thanks! do you know what's the difference between the old influx (which was 0.4?) and the new one or not yet? what will happen to the land fluxes in the long run?

**Comment by oksanaguba** [comment]
by the long run I meant future, not long simulations.

**Comment by golaz** [comment]
@oksanaguba : the land model doesn't know about the water temperature, therefore there should not be a correction for water falling on and evaporating from land. But there should be a correction for water falling on land and running off to the ocean. MPAS-O does include that term.

**Comment by golaz** [comment]
@rljacob : another PR we need for b2beta4. Can this go in?

**Comment by rljacob** [comment]
So this will be BFB?  Sure we can add it tomorrow.  Too late for testing tonight.

**Comment by singhbalwinder** [comment]
Merged to next

**Comment by rljacob** [comment]
@singhbalwinder did you push your copy of next?  I don't see your merge.

**Comment by singhbalwinder** [comment]
Thanks @rljacob for noticing that. I think next was updated before I pushed my changes, so they got rejected. I will try again in a few minutes.

**Comment by singhbalwinder** [comment]
I have now merged this branch with next.

**Comment by golaz** [comment]
@singhbalwinder : is this ready to be merged to master?

**Pull request merged**
